.github/workflows/publish.yml

```diff
@@ -0,0 +1,52 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      # Set up JDK
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      # Set Gradle Properties as Environment Variables
+      # You can use any gradle property in the gradle.properties file as an environment variable.
+      # eg: ballerinaVersion=2.0.0 -> gradle.ballerinaVersion
+      - name: Set Gradle Properties as Environment Variables
```

Comment:
Shall we check the possibility of publishing a timestamp version of the package rather than the SNAPSHOT version. As far as I understood current process publishes a package version like `x.y.z-SNAPSHOT`. But, when we merge multiple PRs before releasing the package to central this can be problematic. Hence, it would great if we could publish a version like `x.y.z-20241119-202600-730b1cd` with a timestamp followed by a random string.

Reply:
Is this correct?
https://github.com/chathushkaayash/ballerina-blocking-queue/blob/7902583f4612df6dffb313e77d6ae842310b8349/.github/workflows/publish.yml#L44C1-L53C80

Reply:
+1